src/template.rs

```diff
@@ -148,29 +151,96 @@ pub(crate) fn walk_dir(
         pbar.set_message(&filename.display().to_string());
 
         if matcher.should_include(relative_path) {
-            let new_contents = engine
-                .clone()
-                .parse_file(filename)?
-                .render(&template)
-                .with_context(|| {
-                    format!(
-                        "{} {} `{}`",
-                        emoji::ERROR,
-                        style("Error replacing placeholders").bold().red(),
-                        style(filename.display()).bold()
-                    )
-                })?;
-            fs::write(filename, new_contents).with_context(|| {
-                format!(
-                    "{} {} `{}`",
-                    emoji::ERROR,
-                    style("Error writing").bold().red(),
-                    style(filename.display()).bold()
-                )
-            })?;
+            let parsed_file = engine.clone().parse_file(filename);
+            match parsed_file {
+                Ok(parsed_file) => parse_and_replace(&template, filename, parsed_file)?,
+                Err(e) => {
+                    if verbose {
+                        files_with_errors.push((filename.display().to_string(), e.clone()));
+                    }
+                }
+            }
         }
     }
+    if files_with_errors.len() > 0 {
+        let warn = construct_substitution_warning(files_with_errors);
+        println!("{}", warn);
+    }
 
     pbar.finish_and_clear();
     Ok(())
 }
+
+fn parse_and_replace(
+    context: &Object,
+    filename: &Path,
+    parsed_file: Template,
+) -> Result<(), anyhow::Error> {
+    let new_content = parse_file(context, filename, &parsed_file);
+    fs::write(filename, new_content).with_context(|| {
+        format!(
+            "{} {} `{}`",
+            emoji::ERROR,
+            style("Error writing").bold().red(),
+            style(filename.display()).bold()
+        )
+    })?;
+
+    Ok(())
+}
+
+fn parse_file(context: &Object, filename: &Path, parsed_file: &Template) -> String {
+    match parsed_file.render(context) {
+        Ok(content) => content,
+        Err(e) => {
+            let msg = e.to_string();
+            if msg.contains("requested variable") {
+                // so, we miss a variable that is present in the file to render
+                let requested_var =
+                    regex::Regex::new(r"(?P<p>.*requested\svariable=)(?P<v>.*)").unwrap();
+                let captures = requested_var.captures(msg.as_str()).unwrap();
+
+                if let Some(Some(req_var)) = captures.iter().last() {
+                    let missing_variable = req_var.as_str().to_string();
+                    // try again with this variable added to the context
+                    let mut context = context.clone();
+                    context.insert(missing_variable.into(), Value::scalar("".to_string()));
```

Comment:
this is something to be discussed, one could argue to keep the variable as it is provided. So better might be to substitute it with something like `format!("{{ {} }}", missing_variable.into())`

Reply:
It might be even possible to promt the user an interactive input for defining this variable. So that the decision is delegated to the user and if he not wants this variable filled he can leave it empty and move on.

@k0pernicus EverlastingBugstopper any opinions on that?